src/openvic-simulation/interface/GFXObject.cpp

```diff
@@ -190,3 +136,178 @@ bool Actor::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
 
 	return ret;
 }
+
+/* arrows.gfx */
+ArrowType::ArrowType() : size { 5 }, texture_file {}, body_texture_file {},
+back_colour {}, progress_colour {}, end_at { 1 }, height { 1 }, arrow_type { 0 },
+heading { 1 }, effect_file {} {}
+
+bool ArrowType::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"size", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(size)),
+		"textureFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_file)),
+		"bodytexture", ONE_EXACTLY, expect_string(assign_variable_callback_string(body_texture_file)),
+		"color", ONE_EXACTLY, expect_colour(assign_variable_callback(back_colour)),
+		"colortwo", ONE_EXACTLY, expect_colour(assign_variable_callback(progress_colour)),
+		"endAt", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(end_at)),
+		"height", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(height)),
+		"type", ONE_EXACTLY, expect_int64(assign_variable_callback(arrow_type)),
+		"heading", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(heading)),
+		"effect", ONE_EXACTLY, expect_string(assign_variable_callback_string(effect_file))
+	);
+
+	return ret;
+}
+
+/* battlearrow.gfx */
+BattleArrow::BattleArrow() : texture_arrow_body {}, texture_arrow_head {}, start { 1 }, stop { 1 },
+ x { 1 }, y { 1 }, font {}, scale { 1 }, no_fade { false }, texture_loop {} {}
+
+bool BattleArrow::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"textureFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_arrow_body)),
+		"textureFile1", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_arrow_head)),
+		"start", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(start)),
+		"stop", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(stop)),
+		"x", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(x)),
+		"y", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(y)),
+		"font", ONE_EXACTLY, expect_string(assign_variable_callback_string(font)),
+		"scale", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(scale)),
+		"nofade", ZERO_OR_ONE, expect_bool(assign_variable_callback(no_fade)),
+		"textureloop", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(texture_loop))
+	);
+
+	return ret;
+}
+
+MapInfo::MapInfo() : texture_file {}, scale { 1 } {}
+
+bool MapInfo::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"textureFile", ZERO_OR_ONE, expect_string(assign_variable_callback_string(texture_file)),
+		"scale", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(scale))
+	);
+
+	return ret;
+}
+
+/* MapItems.gfx */
+Projection::Projection() : 
```

Comment:
```suggestion
Projection::Projection() :
```
Trailing space at the end of this line, same for `Billboard`, `UnitStatsBillboard`, `TextBlock` and `AnimatedMapText`'s constructors